jsonnet/kube-thanos/kube-thanos-receive.libsonnet

```diff
@@ -0,0 +1,94 @@
+local k = import 'ksonnet/ksonnet.beta.4/k.libsonnet';
+
+{
+  _config+:: {
+    receive+: {
+      name: 'thanos-receive',
+      labels: { app: $._config.receive.name },
+      ports: {
+        grpc: 10901,
+        remoteWrite: 19291,
+      },
+    },
+  },
+
+  thanos+:: {
+    receive: {
+      service:
+        local service = k.core.v1.service;
+        local ports = service.mixin.spec.portsType;
+
+        service.new(
+          $._config.receive.name,
+          $._config.receive.labels,
+          [
+            ports.newNamed('grpc', $._config.receive.ports.grpc, $._config.receive.ports.grpc),
+            ports.newNamed('remote-write', $._config.receive.ports.remoteWrite, $._config.receive.ports.remoteWrite),
+          ]
+        ) +
+        service.mixin.metadata.withNamespace($._config.namespace) +
+        service.mixin.metadata.withLabels($._config.receive.labels) +
+        service.mixin.spec.withClusterIp('None'),
+
+      statefulSet:
+        local sts = k.apps.v1.statefulSet;
+        local volume = sts.mixin.spec.template.spec.volumesType;
+        local container = sts.mixin.spec.template.spec.containersType;
+        local containerEnv = container.envType;
+        local containerVolumeMount = container.volumeMountsType;
+
+        local c =
+          container.new($._config.store.name, $._config.images.thanos) +
+          container.withArgs([
+            'receive',
+            '--remote-write.address=0.0.0.0:%d' % $._config.receive.ports.remoteWrite,
+            // '--data-dir=/var/thanos/store',
```

Comment:
we should make the grpc port configurable via a flag, otherwise, someone will change it in `$._config` but it won't actually affect the container.

Reply:
The grpc port is configurable via the service, but not on the container itself. Not sure if we should make this configurable to begin with. :thinking:

Reply:
right, but if someone changes the `_config`, this will make the service point to a port on the container that is not actually open.